tourniquet/patch_lang.py

```diff
@@ -211,6 +197,7 @@ def concretize(self, line: int, col: int, db_context, module_name) -> List[str]:
          second
          [c1d1, c1d2, c1d3]
         """
+        # TODO(ww): Figure out what this is doing.
```

Comment:
@carsonharmon Could you help me understand what this code is doing? I'm finding it a little hard to comprehend.

Reply:
Yes! This is the concretization which adds all the permutations. 

Imagine you have a statement list like this 

ifStmt(...), return (Variable()) 

You first concretize the if statement, and you get a list of strings like this 
["if...x", "if...y"] 
Then you concretize the return statement, which returns another list of strings 
[return a... return b...] 

To create our search space, we have all permutations of these. So you start with whatever list you have, and add the new one to each element 

["if..x return a", "if...x return b", "if... y return a", ... "if.. y return b"] 

So the search space can get quite large, but thats by design

Reply:
Does that make sense?

Reply:
Gotcha. That does make sense. I'll refactor it to actually do the generator in-line, now that I understand it.